### ISSUE TITLE
Fiterrors in MCMCfitter and Eventoptimize

### ISSUE THREAD

**Issue by devbhakt** [nanograv/PINT#1067]
The fiterrs acquired using get_fit_keyvals are not converted properly in certain cases, such as RAJ and DECJ. The fiterrs array has the model errors in houranglesecond and arcseconds instead of hourangle and degree.

**Comment by scottransom** [comment]
Do those errors have units attached, though?  If they do, then they are fine, since you can convert them to whatever you need using Astropy units' `.to()` feature.  If they units are removed, though, then I agree this is a problem!

**Comment by devbhakt** [comment]
Currently, the fiterrs are only taking the value of the uncertainty and the units are removed. My pull request adds in a fix by creating an array for the model fit value units and uses them to convert the uncertainty using Astropy units' `.to()`

**Comment by aarchiba** [comment]
There's a mystery here, one we might should check on with @luojing1211 - why are the units of `model.RAJ.uncertainty.unit != model.RAJ.units`? Are there other kinds of parameter where that occurs? How should the code decide what units the uncertainty should be in?

**Comment by scottransom** [comment]
Yeah, I agree that this is confusing.  @devbhakt and I were just talking about it.  I think the issue is that this is basically what happens in TEMPO/TEMPO2 parfiles.  The positions are specified in full string coords, but the errors are in seconds of time (for RAJ) and arcseconds (for DEC).  Seems like we should always keep the units of the uncertainty the same as the units of the parameters themselves, and change that (if needed, like for RAJ and DECJ) only on outputting to a parfile.
